tests/scribeAuth.test.ts

```diff
@@ -120,5 +230,26 @@ function assertTokens(tokens: Tokens): boolean {
 
 async function getRefreshToken() {
   const tokens = await access.getTokens({ username, password });
-  return tokens.refreshToken;
+  if ('refreshToken' in tokens) {
+    return tokens.refreshToken;
+  }
+  throw new Error('use getRefreshTokenWithMFA() instead');
+}
+
+async function getRefreshTokenWithMFA() {
+  const challenge = await access.getTokens({ username: username2, password });
+  const code = authenticator.generate(otp);
+  if ('user' in challenge && 'challengeParameters' in challenge) {
+    const response = await access.respondToAuthChallengeMfa(
+      challenge.user,
+      code,
+      challenge.challengeParameters
+    );
+    return response.refreshToken;
+  }
+  throw new Error('use getRefreshToken() instead');
+}
+
+function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
 }
```

Comment:
?

Reply:
it's used to create a new code after the last expired

Reply:
you're not allowed to not know about:
```ts
import { setTimeout } from 'node:timers/promises';

await setTimeout(ms)
```